runtime/fiber.c

```diff
@@ -347,7 +347,11 @@ CAMLprim value caml_ensure_stack_capacity(value required_space)
 */
 
 Caml_inline int is_block_and_not_code_frag(value v) {
-  return Is_block(v) && caml_find_code_fragment_by_pc((char *) v) == NULL;
+  if (!Is_block(v)) return 0;
+
+  /* Bytecode code pointers should always be tagged */
+  CAMLassert (caml_find_code_fragment_by_pc((char *) v) == NULL);
+  return 1;
 }
```

Comment:
If I understand correctly, this place is the main change, and to understand why it matters one needs to look at the `caml_scan_stack` code which is used by the GC to find values reachable from a stack:

https://github.com/ocaml/ocaml/blob/58418c71f3eed2201229c91c63b0f5e6744ac648/runtime/fiber.c#L363-L370